assets/javascripts/discourse/initializers/chat-setup.js

```diff
@@ -125,6 +127,157 @@ export default {
         });
       });
     });
+
+    withPluginApi("1.3.0", (api) => {
+      api.addSidebarSection(
+        (BaseCustomSidebarSection, BaseCustomSidebarSectionLink) => {
+          const SidebarChatSectionLink = class extends BaseCustomSidebarSectionLink {
+            constructor({ channel }) {
+              super(...arguments);
+              this.channel = channel;
+            }
+
+            get name() {
+              return this.channel.chatable_id;
+            }
+
+            get route() {
+              return "chat.channel";
+            }
+
+            get model() {
+              return {
+                chatChannel: this.channel,
+                channelId: this.channel.id,
+                channelTitle: this.channel.chatable.slug,
+              };
+            }
+
+            get title() {
+              return this.channel.title;
+            }
+
+            get text() {
+              return this.channel.title;
+            }
+
+            get prefixIcon() {
+              return "hashtag";
+            }
+
+            get prefixIconColor() {
+              return this.channel.chatable.color;
+            }
+
+            get prefixIconBadge() {
+              return this.channel.chatable.read_restricted && "lock";
```

Comment:
The expected return value is a string here so if `read_restricted` is `false`, we might break that expectation.